src/schema-convertors/internalToExpanded.ts

```diff
@@ -4,94 +4,120 @@ import { InternalTypeToStandardTypeMap, RELAXED_EJSON_DEFINITIONS } from './inte
-  if (isSingleType) {
-    return parseType(definedTypes[0], signal);
+  }: TType) {
+    return {
+      ...(typeof hasDuplicates === 'boolean' ? { hasDuplicates } : {}),
+      probability,
+      count
+    };
   }
-  const parsedTypes = await Promise.all(definedTypes.map(type => parseType(type, signal)));
-  return {
-    anyOf: parsedTypes
-  };
-}
 
-async function parseFields(fields: DocumentSchemaType['fields'], signal?: AbortSignal): Promise<{
-  required: ExpandedJSONSchema['required'],
-  properties: ExpandedJSONSchema['properties'],
-}> {
-  const required = [];
-  const properties: ExpandedJSONSchema['properties'] = {};
-  for (const field of fields) {
-    if (field.probability === 1) required.push(field.name);
-    properties[field.name] = {
-      ...await parseTypes(field.types, signal),
-      'x-metadata': getMetadata(field)
+  private async parseTypes(types: SchemaType[], signal?: AbortSignal): Promise<ExpandedJSONSchema> {
+    await allowAbort(signal);
+    const definedTypes = types.filter(type => type.bsonType.toLowerCase() !== 'undefined');
+    const isSingleType = definedTypes.length === 1;
+    if (isSingleType) {
+      return this.parseType(definedTypes[0], signal);
+    }
+    const parsedTypes = await Promise.all(definedTypes.map(type => this.parseType(type, signal)));
+    return {
+      anyOf: parsedTypes
     };
   }
 
-  return { required, properties };
-}
+  private async parseFields(fields: DocumentSchemaType['fields'], signal?: AbortSignal): Promise<{
+    required: ExpandedJSONSchema['required'],
+    properties: ExpandedJSONSchema['properties'],
+  }> {
+    const required = [];
+    const properties: ExpandedJSONSchema['properties'] = {};
+    for (const field of fields) {
+      if (field.probability === 1) required.push(field.name);
+      properties[field.name] = {
+        ...await this.parseTypes(field.types, signal),
+        'x-metadata': this.getMetadata(field)
+      };
+    }
 
-export default async function internalSchemaToMongodb(
-  internalSchema: InternalSchema,
-  options: {
-    signal?: AbortSignal
-} = {}): Promise<ExpandedJSONSchema> {
-  const { required, properties } = await parseFields(internalSchema.fields, options.signal);
-  const schema: ExpandedJSONSchema = {
-    type: 'object',
-    'x-bsonType': 'object',
-    required,
-    properties,
-    $defs: RELAXED_EJSON_DEFINITIONS
-  };
-  return schema;
+    return { required, properties };
+  }
+
+  public async convert(
+    internalSchema: InternalSchema,
+    options: {
+      signal?: AbortSignal
+  } = {}): Promise<ExpandedJSONSchema> {
+    this.clearUsedDefintions();
+    const { required, properties } = await this.parseFields(internalSchema.fields, options.signal);
+    const schema: ExpandedJSONSchema = {
+      type: 'object',
+      'x-bsonType': 'object',
+      required,
+      properties,
+      $defs: this.getUsedDefinitions()
+    };
+    return schema;
+  }
 }
+
+export default InternalToExpandedConvertor;
```

Comment:
Now with index.ts gone, we really don't have a reason to use default exports anymore, do we?